empiric-package/empiric/core/entry.py

```diff
@@ -4,13 +4,25 @@
 
 from empiric.core.utils import felt_to_str, str_to_felt
 
-
-class Entry:
+class BaseEntry:
+    # Make this the generic
+    type: str
```

Comment:
why add a string for the type?

Reply:
(idea was to use this to check membership, e.g. if you hand me a list of Entry objects, I need to filter into separate FutureEntry and SpotEntry lists)